### ISSUE TITLE
parser panics instead of returning an error

### ISSUE THREAD

**Issue by theKlisha** [martinohmann/hcl-rs#284]
Provided code snippet causes panic.

```rs
let str = r#"
  locals {
    test = {
      a = b// this comment breaks the parser
      c = d // but this one doesn't
    }
  }
"#;

let _: Result<hcl::Body, _> = hcl::from_str(str);
```

tested on version `0.16.3`

**Comment by martinohmann** [comment]
Hi there!

Thanks for the bug report and the reproducer.

I'm trying to get this fixed with the next releasebut it may take a couple of days since i just moved to a new home and still need to unpack stuff in order to set up my workspace again 😉.

**Comment by martinohmann** [comment]
@theKlisha Got to it earlier than expected. I just released the fix as 0.16.4.

**Comment by theKlisha** [comment]
Thanks for the fix!
